tv_apps/tv-deck/js/tv_deck.js

```diff
@@ -0,0 +1,538 @@
+/* global TVDeck, ChannelManager, KeyNavigationAdapter, SimpleKeyNavigation */
+
+'use strict';
+
+(function(exports) {
+
+  /**
+   * TVDeck is the app that can watch TV channels by connecting TVManager API.
+   * Every (tuner, source, channel) combination creates an unique URL hash.
+   * Thus, channel switching can be done by simply changing the URL hash and
+   * handled by onhashchange handler.
+   */
+  window.TVDeck = function TVDeck() {
```

Comment:
Would you mind to break the constructor to multiple function and call them at the constructor?